### PR TITLE
CLN fix lr_scheduler+persistent workers in objective+rfc

### PR DESCRIPTION
**Pull request by tomMoral**

### CODE REVIEW

File: utils/torch_solver.py [hunk 1]
```diff
@@ -57,11 +57,12 @@ def set_objective(self, model_init_fn, dataset):
 
         # TODO: num_worker should not be hard coded. Finding a sensible way to
         # set this value is necessary here.
-        system = os.environ.get('RUNNER_OS', platform.system())
-        is_mac = system == 'Darwin' or system == 'macOS'
+        system = os.environ.get('RUNNER_OS', sys.platform)
```
Comment (zaccharieramzi) [inline utils/torch_solver.py:60]: I saw in an SO answer (will find link), that `platform` gives a more reliable system than `sys.platform`. Did you change for a particular reason ?

File: utils/tf_solver.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ def set_objective(self, model_init_fn, dataset):
             ]
         else:
             self.lr_scheduler = lambda epoch: self.lr
-            self.wd_scheduler = lambda epoch: self.decoupled_wd
+            self.wd_scheduler = lambda epoch: self.decoupled_wd * self.lr
```
Comment (zaccharieramzi) [inline utils/tf_solver.py:67]: Yep I don't know how this got through but indeed that's the correct way to scale the weight decay

**Review by zaccharieramzi — APPROVED** [review]
LGTM , small comment on the platform

File: utils/torch_solver.py (hunk 1 above)
Comment (tomMoral) [reply utils/torch_solver.py:60]: No real reason I guess, mostly out of habit.
I found the [SO answer](https://stackoverflow.com/questions/1854/python-what-os-am-i-running-on)

To me, `sys.platform` avoid any dark magic as for `osx` it always returns `darwin` so as it is used under the hood, I would directly use this but I am fine with reverting to `platform`
